allensdk/brain_observatory/behavior/behavior_project_cache/project_apis/data_io/behavior_neuropixels_project_cloud_api.py

```diff
@@ -44,10 +52,16 @@ def get_behavior_session(
         -------
         BehaviorSession
         """
-        row = self._behavior_session_table.query(
-            f"behavior_session_id=={behavior_session_id}"
-        )
-        if row.shape[0] != 1:
+        try:
+            row = self._behavior_session_table.loc[behavior_session_id]
+        except KeyError:
+            raise RuntimeError(
+                "The behavior_session_table should have "
+                "1 and only 1 entry for a given "
+                "behavior_session_id. No sessions found for "
+                f"id={behavior_session_id}"
+            )
+        if len(row.shape) != 1:
```

Comment:
This isn't a very clear check. You are basically checking whether it returns a dataframe (in case of the behavior_session_id repeated) or series in case of behavior_session_id not repeated. 

Could you check the type to see if it's a dataframe or series?

Alternatively, check how many times behavior_session_id appears in the dataframe
```
if (self._behavior_session_table.index == behavior_session_id).sum() != 1:
```

Reply:
Switching to `isintance(obj, pd.Series)`